src/player-service/playerService.ts

```diff
@@ -0,0 +1,80 @@
+import { getRandomCountry } from '../team-service/utils';
+import PlayerRepository from './playerRepository';
+import { v4 as uuidv4 } from 'uuid';
+import { NotFound } from 'http-errors';
+
+class PlayerService {
+
+    constructor(
+        private playerRepository = PlayerRepository,
+    ) { }
+
+    async generatePlayersForTeam(teamId: string) {
+        const positions = {
+            Goalkeeper: 3,
+            Defender: 6,
+            Midfielder: 6,
+            Attacker: 5,
+        };
+
+        const playerPositions: string[] = [];
+
+        for (const [position, count] of Object.entries(positions)) {
+            for (let i = 0; i < count; i++) {
+                playerPositions.push(position);
+            }
+        }
+
+        const players = playerPositions.map((position) => ({
+            id: uuidv4(),
+            team_id: teamId,
+            first_name: `${Math.random().toString(36).substring(7)}`, //ik the names generated will be very weird but it's okay, they can be russian names
+            last_name: `${Math.random().toString(36).substring(7)}`,
+            country: getRandomCountry(),
+            age: Math.floor(Math.random() * 23) + 18,
+            position: position,
+            market_value: '1000000',
+        }));
+
+        return await this.playerRepository.batchCreatePlayers(players);
+    }
+
+    async getPlayersByTeam(teamId: string) {
+        return await this.playerRepository.getPlayersByTeamId(teamId);
+    }
+
+    async placePlayerOnTransferList(playerId: string, askingPrice: string) {
+        const player = await this.playerRepository.findById(playerId);
+        if (!player) throw new NotFound('Player not found');
+
+        await this.playerRepository.update(playerId, { market_value: askingPrice }); //only players with a market value can be placed on the transfer list
+
+        return player;
+    }
+
+    async transferPlayer(playerId: string, newTeamId: string, askingPrice: string) {
```

Comment:
The whole logic of transfer player is missing many points
- Is the user available on the transfer list
- does the user own the team
- does the team has enough money to buy the player 
- how will the user know that this player was on transfer list?

Reply:
- the user isnt available on the list only players.
- yes each user has 1 team which they own
- didnt complete the validation, but tht should be simple check of just checking users money and a players asking price.
- the transfer list has all players tht have an asking price.
-